fe/fe-common/src/main/java/org/apache/doris/catalog/MapType.java

```diff
@@ -192,8 +192,10 @@ protected String prettyPrint(int lpad) {
     }
 
     public static boolean canCastTo(MapType type, MapType targetType) {
-        return Type.canCastTo(type.getKeyType(), targetType.getKeyType())
-            && Type.canCastTo(type.getValueType(), targetType.getValueType());
+        return (targetType.getKeyType().isStringType() && type.getKeyType().isStringType()
```

Comment:
1. add parenthesis for more clear AND OR logic
2. Is already true for StringType canCastTo any type?

Reply:
2. yes , and be will handle this